### ISSUE TITLE
Build Error (TerserWriter)  Cannot read property 'TYPE' of undefined

### ISSUE THREAD

**Issue by rbierbauer** [terser/terser#1492]
When building my app, I get the error

```
Build Error (TerserWriter)

Cannot read property 'TYPE' of undefined
```

according to the logs it happens at `AST_Number.equivalent_to`.

This only happens in the version 5.28.0, it still works when downgrading to 5.27.2.

**Comment by csvan** [comment]
Confirming this as of 5.28, only workaround right now is a downgrade

**Comment by vibin-tv** [comment]
We also facing the same issue when compiling a react application. We directly not using terser as a dependency.
So we have added the following as the fix in package json


  "resolutions": {
    "terser": "5.27.2"
  },

**Comment by SpeckiJ** [comment]
Note: [`npm overrides`](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides)has been introduced with NPM 8.3, these workarounds therefore won't work on NPM versions<8.3

**Comment by fabiosantoscode** [comment]
Fixed in `5.28.1`. Any tips on what input code might be triggering this could be helpful. I still don't know what was wrong with the `5.28.0` release.
